tap_zendesk/__init__.py

```diff
@@ -121,11 +129,27 @@ def do_sync(client, catalog, state, start_date):
 @singer.utils.handle_top_exception(LOGGER)
 def main():
     parsed_args = singer.utils.parse_args(REQUIRED_CONFIG_KEYS)
-
     creds = {
-        "subdomain": parsed_args.config['subdomain'],
-        "oauth_token": parsed_args.config['access_token'],
+        "subdomain": parsed_args.config['subdomain']
     }
+
+    try:
+        oauth_args = singer.utils.parse_args(OAUTH_CONFIG_KEYS)
+        creds.update({
+            "oauth_token": oauth_args.config['access_token'],
+        })
+    except:
+        LOGGER.debug("Cannot find oauth configuration.")
+
+    try:
```

Comment:
I think the default behavior should be to first try oauth and then fallback to api config. So If this `try:` was placed in the `except:` block above that would be accomplished.

We could also take the approach of making the two config styles mutually exclusive, as long as we still favor the oauth style.

Reply:
I don't like nested try/except so I'll find a way to accomplish what you said, which might just be trying to login using OAuth, then trying the API and letting Zenpy raise if need be.